File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingTitleModel.java
package kore.botssdk.models;

import java.io.Serializable;

import kore.botssdk.utils.StringUtils;

public class BrandingTitleModel implements Serializable {
    String name;
    String color;
    String type;
    String img;

    public BrandingTitleModel updateWith(BrandingTitleModel configModel) {
        name = !StringUtils.isNullOrEmpty(configModel.name) ? configModel.name : name;
        color = !StringUtils.isNullOrEmpty(configModel.color) ? configModel.color : color;
        type = !StringUtils.isNullOrEmpty(configModel.type) ? configModel.type : type;
        img = !StringUtils.isNullOrEmpty(configModel.img) ? configModel.img : img;
        return this;
    }

    public String getType() {
        return type;
    }

    public String getImg() {
        return img;
    }

    public String getName() {
        return name;
    }

    public String getColor() {
        return color;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setColor(String color) {
        this.color = color;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingTitleModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class BrandingTitleModel(
    @SerializedName("name")
    val name: String? = null,
    @SerializedName("color")
    var color: String? = null,
    @SerializedName("type")
    val type: String? = null,
    @SerializedName("img")
    val img: String? = null
) : Serializable

fun BrandingTitleModel.updateWith(configModel: BrandingTitleModel): BrandingTitleModel {
    return this.copy(
        name = configModel.name?.ifEmpty { this.name },
        color = configModel.color?.ifEmpty { this.color },
        type = configModel.type?.ifEmpty { this.type },
        img = configModel.img?.ifEmpty { this.img },
    )
}